starknet_py/net/client_models.py

```diff
@@ -1106,3 +1093,85 @@ class BlockTransactionTrace:
 
     transaction_hash: int
     trace_root: TransactionTrace
+
+
+@dataclass
+class BinaryNode:
+    """
+    Dataclass representing an internal node whose both children are non-zero.
+    """
+
+    left: int
+    right: int
+
+
+@dataclass
+class EdgeNode:
+    """
+    Dataclass representing a path to the highest non-zero descendant node.
+    """
+
+    path: int
+    length: int
+    child: int
+
+
+MerkleNode = Union[BinaryNode, EdgeNode]
+
+
+@dataclass
+class NodeHashToNodeMappingItem:
+    node_hash: int
+    node: MerkleNode
+
+
+NodeHashToNodeMapping = List[NodeHashToNodeMappingItem]
+
+
+@dataclass
+class ContractStorageKeys:
+    """
+    Dataclass representing a pair of contract address and storage keys.
+    """
+
+    contract_address: int
+    storage_keys: List[int]
+
+
+@dataclass
+class ContractLeafData:
+    nonce: int
+    class_hash: int
+
+
+@dataclass
+class GlobalRoots:
+    contracts_tree_root: int
+    classes_tree_root: int
+    block_hash: int
+
+
+@dataclass
+class ContractsProof:
+    nodes: NodeHashToNodeMapping
+    contract_leaves_data: List[ContractLeafData]
+    contracts_storage_proof: NodeHashToNodeMapping
+
+
+@dataclass
+class StorageProofResponse:
+    """
+    Dataclass representing a response to a storage proof request.
+    """
+
+    classes_proof: NodeHashToNodeMapping
+    contracts_proof: ContractsProof
+    contracts_storage_proofs: List[NodeHashToNodeMapping]
+    global_roots: GlobalRoots
+
+
+@dataclass
+class MessageStatus:
+    transaction_hash: int
+    finality_status: TransactionFinalityStatus
```

Comment:
https://github.com/starkware-libs/starknet-specs/blob/v0.8.0/api/starknet_api_openrpc.json#L279
```suggestion
    finality_status: TransactionStatus
```